shared/config/config.go

```diff
@@ -60,30 +65,54 @@ type OrderSettlerConfig struct {
-	GasTokenDecimals                uint8            `yaml:"gas_token_decimals"`
-	NumBlockConfirmationsBeforeFill int64            `yaml:"num_block_confirmations_before_fill"`
-	HyperlaneDomain                 string           `yaml:"hyperlane_domain"`
-	QuickStartNumBlocksBack         uint64           `yaml:"quick_start_num_blocks_back"`
-	MinFillSize                     *big.Int         `yaml:"min_fill_size"`
-	MaxFillSize                     *big.Int         `yaml:"max_fill_size"`
-	FastTransferContractAddress     string           `yaml:"fast_transfer_contract_address"`
-	SolverAddress                   string           `yaml:"solver_address"`
-	USDCDenom                       string           `yaml:"usdc_denom"`
-	Relayer                         RelayerConfig    `yaml:"relayer"`
-
+	// e.g. osmosis
+	ChainName string `yaml:"chain_name"`
+	// e.g. osmosis-1
+	ChainID string `yaml:"chain_id"`
+	// (cosmos, evm)
+	Type ChainType `yaml:"type"`
+	// Environment specifies whether this is a mainnet or testnet configuration
+	Environment ChainEnvironment `yaml:"environment"`
+	// Cosmos contains specific configuration for Cosmos-based chains
+	Cosmos *CosmosConfig `yaml:"cosmos,omitempty"`
+	// EVM contains specific configuration for Ethereum Virtual Machine based chains
+	EVM *EVMConfig `yaml:"evm,omitempty"`
+	// GasTokenSymbol is the symbol of the native gas token (e.g., "ETH", "MATIC")
+	GasTokenSymbol string `yaml:"gas_token_symbol"`
+	// GasTokenDecimals specifies the number of decimal places for the gas token
+	GasTokenDecimals uint8 `yaml:"gas_token_decimals"`
+	// NumBlockConfirmationsBeforeFill is the number of block confirmations required
+	// before the solver will attempt to fill an order
+	NumBlockConfirmationsBeforeFill int64 `yaml:"num_block_confirmations_before_fill"`
+	// HyperlaneDomain is the unique identifier for this chain in the Hyperlane
+	// cross-chain messaging system
+	HyperlaneDomain string `yaml:"hyperlane_domain"`
+	// QuickStartNumBlocksBack specifies how many blocks back to start scanning
+	// from when the solver is initialized
+	QuickStartNumBlocksBack uint64 `yaml:"quick_start_num_blocks_back"`
+	// MinFillSize is the minimum amount of USDC that can be processed in a single
+	// order fill. Orders below this size will be abandoned
+	MinFillSize big.Int `yaml:"min_fill_size"`
+	// MaxFillSize is the maximum amount of USDC that can be processed in a single
+	// order fill. Orders exceeding this size will be abandoned
+	MaxFillSize big.Int `yaml:"max_fill_size"`
+	// FastTransferContractAddress is the address of the Skip Go Fast Transfer
+	// Protocol contract deployed on this chain
+	FastTransferContractAddress string `yaml:"fast_transfer_contract_address"`
+	// SolverAddress is the address of the solver wallet on this chain that will
+	// be used to fulfill orders and receive fees
+	SolverAddress string `yaml:"solver_address"`
+	// USDCDenom is the denomination or contract address for USDC on this chain
```

Comment:
could it also be a native token if on Noble?

Reply:
yeah it would still be the ibc denom in this case I belive, `uusdc`

Reply:
I believe this would be native denom then